.github/workflows/test_deploy.yml

```diff
@@ -157,12 +161,30 @@ jobs:
           cd android
           ./gradlew assembleRelease
       
+      - name: Sign Android release
+        # ID used to access signed output path in next step
+        id: sign_app 
+        uses: r0adkll/sign-android-release@v1
+        with:
+          releaseDirectory: ${{ env.ANDROID_RELEASE_DIR }}
+          signingKeyBase64: ${{ secrets.ANDROID_RELEASE_KEYSTORE }}
+          alias: ${{ secrets.ANDROID_RELEASE_KEYSTORE_ALIAS }}
+          keyStorePassword: ${{ secrets.ANDROID_RELEASE_KEYSTORE_STORE_PASSWORD }}
+          keyPassword: ${{ secrets.ANDROID_RELEASE_KEYSTORE_KEY_PASSWORD }}
+        env:
+          # override default build-tools version (29.0.3) -- optional
+          BUILD_TOOLS_VERSION: "30.0.2"
+
+      # The sign_app step returns weird file name app-release-unsigned-signed.apk, rename it to a consistent and reusable name
+      - name: rename signed file
+        run: mv ${{steps.sign_app.outputs.signedReleaseFile}} ${{ env.ANDROID_RELEASE_DIR }}/${{ env.ANDROID_SIGNED_NAME }}
+
```

Comment:
Keystore is now removed from source code and stored in GitHub secrets, this new step is added to use GH secrets to build app.